java/arcs/core/storage/StorageKeyFactory.kt

```diff
@@ -52,8 +52,9 @@ abstract class StorageKeyFactory(val protocol: String, val capabilities: Capabil
         override val arcId: ArcId,
         override val entitySchema: Schema
     ) : StorageKeyOptions {
-        override val unique: String =
-        requireNotNull(entitySchema).name?.name ?: entitySchema.hash
+        override val unique: String = with(entitySchema.name?.name) {
```

Comment:
when will `entitySchema.name` be null? 
also, could you, please, add tests to cover all these edge cases? 
thanks.

Reply:
Also, can we understand what is happening first before making changes to code?

Do we know why it's null? Do we know if this fixes Satakshis problem?

Reply:
I have a few suspicions as to why this fix should work and what is going on. I'm currently testing them out in cider. Currently, it seems like this will fix the issue -- but is it the right fix? I'll keep digging.  

@mmandlis  `entitySchema.name` can be null anytime we have a schema without any names. For instance, 
```
particle TimestampMsToDayRedactor in '.TimestampMsToDayRedactor'
  input: reads [~a with {timestampInMs: Number}]
```
Here, the schema.name should be null. 

Test cases added in 2860f0f.